### PR TITLE
Whisper inference

### PR DESCRIPTION
**Pull request by fxmarty**
As per title - tested in fp16 for now.

This allows to use Transformers Whisper checkpoints with TRT-LLM inference.

Left to do:
- [x] Punctuation sometimes does not match, see https://huggingface.slack.com/archives/C0435BHN1L6/p1712160819124509, have to see if this is a bug or just a numerical artifact (does not impact WER calculated without punctuation), e.g. `test_generation[None-openai/whisper-large-v3]`
- [ ] quality workflow somehow only fails on GA but not locally

### CODE REVIEW

File: examples/automatic-speech-recognition/whisper.py [hunk 1]
```diff
@@ -53,6 +57,30 @@
...
     model.save_pretrained(args.output)
     print(f"TRTLLM engines have been saved at {args.output}.")
+
+    model = WhisperForConditionalGeneration.from_pretrained(args.output)
+    processor = AutoProcessor.from_pretrained(
+        args.model
+    )  # TODO: save processor in args.output (like ORTModel)
+
+    data = load_dataset(
+        "hf-internal-testing/librispeech_asr_dummy", "clean", split="validation"
```
Comment (mfuntowicz) [inline examples/automatic-speech-recognition/whisper.py:71]: Maybe let's use something not internal?

File: src/optimum/nvidia/builder/config.py [hunk 2]
```diff
@@ -269,3 +268,4 @@ def build(self) -> EngineConfig:
             generation_profile=self._generation_profile,
             plugins_config=self._plugin_config,
         )
+        return engine_config
```
Comment (mfuntowicz) [inline src/optimum/nvidia/builder/config.py:271]: Is this change mandatory?

File: src/optimum/nvidia/builder/local.py [hunk 3]
```diff
@@ -120,15 +121,22 @@ def build(self, config: EngineConfig):
...
+                stdout=PIPE,
+                stderr=STDOUT,
+            )
+        LOGGER.info(
+            f"trtllm-build stdout: {result.stdout.decode('utf-8') if result.stdout is not None else None}"
+        )
+        LOGGER.info(
+            f"trtllm-build stderr: {result.stderr.decode('utf-8') if result.stderr is not None else None}"
         )
+
```
Comment (mfuntowicz) [inline src/optimum/nvidia/builder/local.py:138]: Let's print only if result != 0, wdyt?

File: src/optimum/nvidia/utils/patching.py
Comment (mfuntowicz) [inline src/optimum/nvidia/utils/patching.py:1]: Hacky 😂

**Review by mfuntowicz — APPROVED** [review]
LGTM ! A few comments, feel free to merge 😊

File: examples/automatic-speech-recognition/whisper.py (hunk 1 above)
Comment (fxmarty) [reply examples/automatic-speech-recognition/whisper.py:71]: This one is public, it is fine. This is the same in Transformers doc example.

**Pull request merged**